webapp/src/components/App/Studies/StudyTree/index.tsx

```diff
@@ -0,0 +1,298 @@
+        return [accTree, [...accFailedPaths, path]];
+      }
+    },
+    Promise.resolve<[StudyTreeNode, string[]]>([studiesTree, []]),
+  );
+}
+
+/**
+ * Insert a workspace into the study tree if it doesn't exist already.
+ *
+ * @param workspace key of the workspace
+ * @param stydyTree study tree to insert the workspace into
+ * @returns study tree with the empty workspace inserted if it wasn't already there.
+ */
+function insertWorkspaceIfNotExist(
+  stydyTree: StudyTreeNode,
+  workspace: string,
+) {
+  const emptyNode = { name: workspace, path: `/${workspace}`, children: [] };
+  if (stydyTree.children.some((child) => child.name === workspace)) {
+    return stydyTree;
+  }
+  return {
+    ...stydyTree,
+    children: [...stydyTree.children, emptyNode],
+  };
+}
+
+/**
+ * Insert several workspaces into the study tree if they don't exist already in the tree.
+ *
+ * @param workspaces workspaces to insert into the tree
+ * @param stydyTree study tree to insert the workspaces into
+ * @returns study tree with the empty workspaces inserted if they weren't already there.
+ */
+export function insertWorkspacesIfNotExist(
+  stydyTree: StudyTreeNode,
+  workspaces: string[],
+): StudyTreeNode {
+  return workspaces.reduce((acc, workspace) => {
+    return insertWorkspaceIfNotExist(acc, workspace);
+  }, stydyTree);
+}
+
+/**
+ * fetch and insert the workspaces into the study tree.
+ * @param studyTree study tree to insert the workspaces into
+ * @returns study tree with the workspaces inserted if they weren't already there.
+ */
+async function fetchAndInsertWorkspaces(
+  studyTree: StudyTreeNode,
+): Promise<StudyTreeNode> {
+  const workspaces = await api.getWorkspaces();
+  return insertWorkspacesIfNotExist(studyTree, workspaces);
+}
+
+function StudyTree() {
+  const initialStudiesTree = useAppSelector(getStudiesTree);
+  const [studiesTree, setStudiesTree] = useState(initialStudiesTree);
+  const folder = useAppSelector((state) => getStudyFilters(state).folder, R.T);
+  const enqueueErrorSnackbar = useEnqueueErrorSnackbar();
+  const dispatch = useAppDispatch();
+
+  // Initialize folders once we have the tree
+  useUpdateEffectOnce(() => {
+    fetchAndInsertWorkspaces(initialStudiesTree)
+      .then((tree) => {
+        // once we get the workspaces we intialize the first level of folders
+        const chidrenPaths = tree.children.map((child) => `root${child.path}`);
+        fetchAndInsertSubfolders(chidrenPaths, tree).then((r) => {
+          setStudiesTree(r[0]);
+          for (const path of r[1]) {
+            enqueueErrorSnackbar(
+              `Failed to initialize folders for : ${path}`,
+              "details in console.error",
+            );
+          }
+        });
+      })
+      .catch((error) => {
+        enqueueErrorSnackbar("Failed to load list workspaces", error);
+        setStudiesTree(initialStudiesTree);
+      });
+  }, [initialStudiesTree]);
```

Comment:
Refactor folder initialization to use async/await instead of nested promise chains

please consider avoiding promise chaining (unless special case where its needed) because it can lead to the "callback hell" very know issue of nested .then() handlers that make the flow hard to follow:
fetchWorkspaces()
 .then(tree => {
   fetchSubfolders()
     .then(result => {
       handleErrors()
     })
 })

Also as stated in previous review please try to avoid iterators over native methods (unless specific use case) Instead consider:

```typescript
  useUpdateEffectOnce(() => {
    const initializeFolders = async () => {
      try {
        const treeWithWorkspaces =
          await fetchAndInsertWorkspaces(initialStudiesTree);
        const childrenPaths = treeWithWorkspaces.children.map(
          (child) => `root${child.path}`,
        );

        const [updatedTree, failedPaths] = await fetchAndInsertSubfolders(
          childrenPaths,
          treeWithWorkspaces,
        );

        setStudiesTree(updatedTree);
        failedPaths.forEach((path) => {
          enqueueErrorSnackbar(
             // use dedicated translation key here
          );
        });
      } catch (error) {
        enqueueErrorSnackbar(
          // use dedicated translation key here
          error as AxiosError,
        );
        setStudiesTree(initialStudiesTree);
      }
    };

    initializeFolders();
  }, [initialStudiesTree]);
  ```